src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -352,6 +352,7 @@ private bool TryValidateAuthorityAndHost(out string hostText)
             return true;
         }
 
+        [SkipLocalsInit]
         private bool TryValidatePath(ReadOnlySpan<char> pathSegment)
         {
             // Must start with a leading slash
```

Comment:
While you're on it, can you please write the next line as
```c#
if (pathSegment.IsEmpty || pathSegment[0] != '/')
```
Because of the way `IsEmpty` is written, the bound check for `pathSegment[0]` is [elided by the JIT](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdwCBEkMAggAo2MbABMA8l0kBPWWOxcADxgABbYUAB8DP4YIfIA5gI8AJSm5ibU5llMAOzR2LEJSRgsADIwXPGxDAC8NQyMhIT5hTCJFRgA2nQAugwAhPUA5AD0QxqZ5gC+aWazFgxWNgwAQk4uHl6+/kGh4VExcW3FqZNmGdnmxHmHRR0sAJK4AKLiGN4MTS1H7TzdfYMGKNxvMZtQpkA).